modules/core/src/weaver/suites.scala

```diff
@@ -81,37 +80,63 @@ abstract class MutableFSuite[F[_]] extends RunnableSuite[F]  {
     def apply(run : (Res, Log[F]) => F[Expectations]) : Unit = registerTest(name)(res => Test(name.name, log => run(res, log)))
   }
 
+  private[this] var testSeq = Seq.empty[(TestName, Res => F[TestOutcome])]
+
+  def plan: List[TestName] = testSeq.map(_._1).toList
+
+  private[this] var isInitialized = false
+
+  private[this] def initError() =
+    new AssertionError(
+      "Cannot define new tests after TestSuite was initialized"
+    )
+
+
+  private[weaver] def getTests(args: List[String]) = {
```

Comment:
This stuff is just abstracted back up to the common base trait between forEach and standard suite, no changes made.